host/sgx/loadelf.c

```diff
@@ -248,34 +289,26 @@ static oe_result_t _load_elf_image(const char* path, oe_enclave_image_t* image)
         {
             case PT_LOAD:
             {
-                /* Cache the segment properties used during the image load */
-                seg->memsz = ph->p_memsz;
-                seg->vaddr = ph->p_vaddr;
+                /* Cache the segment properties for enclave page add */
+                segment->memsz = ph->p_memsz;
+                segment->vaddr = ph->p_vaddr;
+                segment->flags = ph->p_flags;
 
-                /* Map the segment permission flags to OE equivalents */
-                {
-                    if (ph->p_flags & PF_R)
-                        seg->flags |= OE_SEGMENT_FLAG_READ;
```

Comment:
It appears that this mapping is not there in the new changes. Is it not necessary?

Reply:
I see that this is addressed later. Closing.